src/services/oktaSessionService.ts

```diff
@@ -0,0 +1,34 @@
+import CreateSessionCommand from "../commands/session/createSessionCommand";
+import { SESSIONS_ENDPOINT } from "../config";
+import ConnectorInterface, {
+    ConnectorResponse
+} from "../network/config/connector";
+
+interface SessionServiceInterface {
+    createSession: (
+        createSessionCommand: CreateSessionCommand
+    ) => Promise<ConnectorResponse>;
+}
+
+export default class OktaSessionService implements SessionServiceInterface {
+    private readonly restConnector: ConnectorInterface;
+
+    constructor(restConnector: ConnectorInterface) {
+        if (!restConnector) {
+            throw new Error("No valid REST connector.");
+        }
+
+        this.restConnector = restConnector;
+    }
+
+    public async createSession(
```

Comment:
most of the services methods return a `Promise< ConnectorResponse>` but ConnectorResponse is an interface used between the service and the restConnector. If we return a ConnectorResponse, the consumer of this methd will have to "guess" the type of the returned object. Let's do a typecast here from ConnectorResponse to `CreateSessionResponse`

Reply:
could you also make the change in all other services?

Reply:
Yes !